baxter_collaboration_lib/include/robot_interface/gripper.h

```diff
@@ -18,10 +18,11 @@ class Gripper
     bool   use_robot;       // Flag to know if we're going to use the robot or not
     bool   first_run;       // Flag to calibrate the gripper at startup if needed
 
-    ros::NodeHandle rnh;      // ROS node handle
-    ros::Subscriber sub;      // Subscriber to receive the state of the gripper
-    ros::Publisher  pub;      // Publisher for requesting actions to the gripper
-    ros::Subscriber sub_prop; // Subscriber to receive the properties of the gripper
+    ros::NodeHandle rnh;       // ROS node handle
+    ros::Subscriber sub;       // Subscriber to receive the state of the gripper
+    ros::Publisher  pub;       // Publisher for requesting actions to the gripper
+    ros::Subscriber sub_prop;  // Subscriber to receive the properties of the gripper
+    ros::AsyncSpinner spinner; // AsyncSpinner to handle callbacks
```

Comment:
however, it is giving segfault and double-free corruption errors when exiting. looking into it